### PR TITLE
Added Exactseek Module

### LINKED ISSUE #219: [New Feature] Adding exactseek.com
Here's a simple query:
https://www.exactseek.com/cgi-bin/search.cgi?q=security
Website claims on giving good seo based results.
Looks the contrary, extremely helpful for OSINT

### PR DESCRIPTION
**Pull request by rachejazz**
Fixes #219 

#### Changes

- Reverted activesearch changes (as said by @k4u5h1k )
- Added exactseek module
- Added exactseek util

#### Checklist

- [x] I have read the Contribution & Best practices Guideline.
- [x] My branch is up-to-date with the Upstream `master` branch.
- [ ] The acceptance, integration, unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by rachejazz** [comment]
```
┬─[divya at racharch in ~/a/g/OWASP-Maryam]─[G:search]
╰──> λ ./maryam -e exactseek -q passwords
[*] [EXACTSEEK]Searching in exactseek.com...
[*] hashcat - advanced password recovery
World's fastest and most advanced password recovery utility ... Supports reading password candidates from file and stdin. *Supports hex-salt and hex- ... Supports password candidate brain functionality. *Supports distributed cracking networks ... Blo
Details: hashcat.net/

[*] NirSoft - freeware utilities: password recovery, system utilities, desktop...
Password Recovery Utilities *WebBrowserPassView - View the passwords stored by your Web  ... Dialupass: Dialup Password Recovery - Recovers the passwords of dialup entries (VPN and  ... Network Password Recovery - Freeware utility that recovers the n
Details: nirsoft.net/

[*] Password Manager for Families, Businesses, Teams | 1Password
1Password remembers all your passwords for you to help keep account information safe. ... A password manager, digital vault, form filler and secure digital wallet.  ... The secure enterprise password manager. More than 60,000 businesses trust 1Passwo
Details: 1password.com/

[*] Password Manager App for Home, Mobile, Business | Dashlane
Store and fill in all your passwords, personal information, and payments. ... Your passwords should be, too. Dashlane creates, saves, and autofills strong, randomized  ...
Details: dashlane.com/

[*] How Secure Is My Password?
How long it would take a computer to crack your password? ...
Details: howsecureismypassword.net/

[*] Digital Forensic, Data Decryption and Password Recovery Solutions for Law...
Try professional password recovery, data decryption, mobile and cloud forensic tools from ... Elcomsoft Internet Password Breaker 3.30 extracts passwords from Tor and 360 Safe  ... Elcomsoft password recovery solutions allow gaining access to passwor
Details: elcomsoft.com/

[*] 1 Password Manager & Vault App, Enterprise SSO & MFA | LastPass
Generate strong passwords and store them in a secure vault. Now with enterprise SSO and  ... Generate strong passwords. The built-in password generator creates long, randomized  ... Save a password once, and it's instantly available on all your devic
Details: lastpass.com/

[*] Strong Random Password Generator
...  and learn over 30 tricks to keep your passwords, accounts and documents safe. ... Strong Password Generator to create secure passwords that are impossible to crack on your ... Remember your password:. Remember your password with the first charac
Details: passwordsgenerator.net/

[*] FIDO Alliance - Open Authentication Standards More Secure than Passwords
...  open and free authentication standards to help reduce the world's reliance on passwords,  ... A global collaboration driving the world to move beyond passwords. More Perspectives All  ... Here's why Google, Apple and Microsoft think it's time to
Details: fidoalliance.org/

[*] Bitwarden Open Source Password Manager | Bitwarden
...  makes it easy to generate and store unique passwords for any browser or device. Create  ... Secure Your Passwords From Any Location or Device. The Bitwarden Password Manager helps  ... Secure Password Generator. Encrypted file storage. -. 1 GB P
Details: bitwarden.com/

[*] Thinfi | Password protect a short URL
With thinfi you can convert Internationalized Domain Names and password protect (optional ... Create a short URL with password protection ...
Details: thinfi.com/

[*] TeamsID Business Password Manager | Access Security‎ Solutions
Try TeamsID Business Password Manager FREE for 14 days! ... Password Manager drastically improves the security of an organization's passwords,  ... TeamsID Business Password Manager. The TeamsID Business Password Manager will drastically ... Secret P
Details: teamsid.com/

[*] Keeper Security | Best Personal and Business Password Manager
Keeper is the top-rated personal and business password manager for protection from  ... password-related data breaches and cyberthreats. Try free now! ... Cybersecurity starts with password security.. Keeper is the top-rated password manager  ... Kee
Details: keepersecurity.com/

[*] Personal Website of Kieran O'Shea
password = 'password';. $hash = phpbb_hash($password);. if (phpbb_check_hash($password, $ ... echo '"Hash "'.$hash.'" matches password "'.$password.'"';. }. It allows you to run a  ... Hashing phpBB Passwords. April 15, 2018 @ 4:31 pm · Filed under C
Details: kieranoshea.com/

[*] World Scientific Publishing Co Pte Ltd
Can't sign in? Forgot your password?. Enter your email address below and we will send you ... Please check your inbox for the reset password link that is only valid for 24 hours. ... an existing account you will receive an email with instructions to 
Details: worldscinet.com/

[*] Greyhound
Reset your password. We will send you a link to reset your password to the below email,  ... Reset your password. To reset your password simply complete the fields below. ... Reset your password. To reset your password please complete the following s
Details: greyhound.com/

[*] WatchGuard: Network Security, Secure Wi-Fi, and MFA Solutions
Weak and Stolen Passwords. Mounting pressure to devise strong, unique passwords, every 90 ... serialize and share passwords between accounts for even less password security than  ... hardware tokens are sealed electronic devices that generate secure 
Details: watchguard.com/

[*] Core77 / Industrial Design Magazine + Resource
Reset Password. Please enter your email and we will send an email to reset your password. ...
Details: core77.com/

[*] DWUser.com - Dreamweaver Extensions, Slideshow and Media Presentation Creator,...
Password:. Reset Password.... Why create an account? *Manage support tickets and  ...
Details: dwuser.com/

[*] CGSecurity
Password recovery. BIOS setup can be password protected. You can get back your password  ...
Details: cgsecurity.org/

[*] Home | MIT Press Journals
Can't sign in? Forgot your password?. Enter your email address below and we will send you ... an existing account you will receive an email with instructions to reset your password. ...
Details: mitpressjournals.org/

[*] Features and Tools | Vimeo
Password protection. Keep your videos private with custom password protection.. * Video  ...
Details: player.vimeo.com/

[*] Mary Ann Liebert, Inc. | Home
Can't sign in? Forgot your password?. Enter your email address below and we will send you ... an existing account you will receive an email with instructions to reset your password. ...
Details: liebertpub.com/

[*] Home | Spirituality & Health
If you haven't done so already, please create a new password here. ...
Details: spiritualityhealth.com/

[*] Homepage | ConnectSafely
...  and what's wrong with passwords. "Simply put, passwords aren't up to the task." ... ... and what's wrong with passwords. "Simply put, passwords aren't up to the task." ... ... Do Passwords Really Protect Us?. October 5, 2020. Online security exp
Details: connectsafely.org/

[*] The FASEB Journal - Wiley Online Library
Please check your email for instructions on resetting your password. If you do not  ...
Details: fasebj.org/

[*] Health Affairs
Can't sign in? Forgot your password?. Enter your email address below and we will send you ... an existing account you will receive an email with instructions to reset your password. ...
Details: healthaffairs.org/

[*] FileFactory
Password protect files. Keep your files safe! Password protect files and folders easily.  ... Only the person who knows the password will be able to download your files. ...
Details: filefactory.com/

[*] Home - Embedded.com
Create New Password. We've sent you an email with instructions to create a new password.  ... Create New Password. We've sent an email with instructions to create a new password. Your ... Create New Password. We'll send you a link to create a new pas
Details: embedded.com/

[*] Star Trek (Official Site)
Confirm Password. Your password must include at least 8 characters with a combination of  ... Forgot Your Password?. Enter the email address associated with your account and we'll  ...
Details: startrek.com/
```

File: core/util/exactseek.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+		self.q = q
+		self.limit = limit
+		self.framework = main.framework
+		self._pages = ''
+		self._json = {}
+		self.ex = 'https://www.exactseek.com/cgi-bin/search.cgi'
+
+	def run_crawl(self):
+		self.framework.verbose("[EXACTSEEK]Searching in exactseek.com...")
+		url = f"{self.ex}?q={self.q}"
```
Comment (saeeddhqan) [inline core/util/exactseek.py]: add self.q as a params parameter.

File: core/util/exactseek.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+		"""
+		self.q = q
+		self.limit = limit
+		self.framework = main.framework
+		self._pages = ''
+		self._json = {}
+		self.ex = 'https://www.exactseek.com/cgi-bin/search.cgi'
+
+	def run_crawl(self):
+		self.framework.verbose("[EXACTSEEK]Searching in exactseek.com...")
```
Comment (saeeddhqan) [inline core/util/exactseek.py]: use ' .. '

File: core/util/exactseek.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+		url = f"{self.ex}?q={self.q}"
+		max_page = 1
+		page_no = 0
+		while(int(max_page) > page_no and page_no != int(self.limit)):
```
Comment (saeeddhqan) [inline core/util/exactseek.py]: self.limit isn't an int?

File: core/util/exactseek.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+			try:
+				req = self.framework.request(
+					url = url,
+					params = {'s':f'{page_no}1'}
+				)
+				try:
```
Comment (saeeddhqan) [inline core/util/exactseek.py]: here, don't use try. instead, use if/else

**Comment by saeeddhqan** [comment]
any new search source should be like dogpile, google, yahoo style. So, use lxml instead of re. 
you can do it any time. here is the matter of format.

**Comment by rachejazz** [comment]
@saeeddhqan I'll do it the next time I add new sources.

**Comment by rachejazz** [comment]
@saeeddhqan i just now saw yahoo and google is using re as well. Where exactly did you say about formatting?

**Comment by saeeddhqan** [comment]
yahoo and google are using re as well: no. yahoo just uses re for minor things. google doesn't use re at all.

**Pull request closed without merging**